x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -0,0 +1,246 @@
+  const [EMSSuggestions, setEMSSuggestions] = useState<
+    Array<MLEMSTermJoinConfig | null> | undefined
+  >();
+  const {
+    services: { maps: mapsPlugin },
+  } = useMlKibana();
+
+  const getEMSTermSuggestions = useCallback(async (): Promise<void> => {
+    if (mapsPlugin) {
+      const suggestions = await Promise.all(
+        jobIds.map(async (jobId) => {
+          const entityValues = new Set<string>();
+          let entityName;
+          for (let i = 0; i < anomalies.length; i++) {
+            if (jobId === anomalies[i].jobId && anomalies[i].entityValue !== '') {
+              entityValues.add(anomalies[i].entityValue);
+
+              if (!entityName) {
+                entityName = anomalies[i].entityName;
+              }
+            }
+            if (
+              // convert to set so it's unique values
+              entityValues.size === MAX_ENTITY_VALUES ||
+              (anomalies.length > MAX_ENTITY_VALUES && i > Math.floor(anomalies.length / 3))
+            )
+              break;
+          }
+
+          const suggestion: EMSTermJoinConfig | null = await mapsPlugin.suggestEMSTermJoinConfig({
+            emsLayerIds: COMMON_EMS_LAYER_IDS,
+            sampleValues: Array.from(entityValues),
+            sampleValuesColumnName: entityName || '',
+          });
+          if (suggestion) {
+            return { jobId, ...suggestion };
+          }
+          return suggestion;
+        })
+      );
+
+      setEMSSuggestions(suggestions.filter((s) => s !== null));
+    }
+  }, [...jobIds]);
+
+  useEffect(
+    function getInitialEMSTermSuggestions() {
+      if (anomalies && anomalies.length > 0) {
+        getEMSTermSuggestions();
+      }
+    },
+    [...jobIds]
+  );
+
+  useEffect(
+    function setupMapLayers() {
+      // Loop through suggestions list and make a layer for each
+      if (EMSSuggestions?.length) {
+        let count = 0;
+        const layers = EMSSuggestions.reduce(function (result, suggestion) {
+          if (suggestion) {
+            const visible = count === 0;
+            result.push(getChoroplethAnomaliesLayer(anomalies, suggestion, visible));
+            count++;
+          }
+          return result;
+        }, [] as VectorLayerDescriptor[]);
+
+        setLayerList(layers);
+      }
+    },
+    [EMSSuggestions, anomalies.length]
+  );
+
+  if (EMSSuggestions?.length === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiPanel data-test-subj="mlAnomalyExplorerAnomaliesMap">
+        <EuiAccordion
+          id="mlAnomalyExplorerAnomaliesMapAccordionId"
+          initialIsOpen={true}
+          buttonContent={
+            <EuiTitle className="panel-title">
+              <h2>
+                <FormattedMessage id="xpack.ml.explorer.mapTitle" defaultMessage="Anomaly Map" />
+              </h2>
+            </EuiTitle>
+          }
+        >
+          <div
+            data-test-subj="xpack.ml.explorer.anomaliesMap"
```

Comment:
nit: I think we follow another naming pattern for naming `data-test-subj`, e.g. `mlExplorernomaliesMap`

Reply:
Updated in d92e417a9fe57db1172fb65ad003fb6742be1ff6